CCallback.cpp

```diff
@@ -217,6 +217,16 @@ bool CCallback::buildBuilding(const CGTownInstance *town, BuildingID buildingID)
 	return true;
 }
 
+bool CCallback::triggerTownSpecialBuildingAction(const CGTownInstance *town, BuildingSubID::EBuildingSubID subBuildingID)
```

Comment:
We might also want to extend this logic to mana whirlpool, similar to what hota did with it.
Or even allow it for any building, e.g. flag in building config, so you can make explicitly visitable stables if you want to

Reply:
Yes, there's nothing wrong with extending it?

Reply:
Yeah, should be fine allowing this for all buildings.

Explicit visit can also be used to fix our old feature-bug where stables & such are visited on new week automatically, without even opening town screen

Reply:
well there are 3 types of visit that make sense:
1. automatic (like now) - it is kinda QoL that buildings like stables work without opening town screen
2. on opening town screen (original h3 mana vortex)
3. hota 1.7 has its new manual handling of mana vortex

Reply:
> automatic (like now) - it is kinda QoL 

Yeah, but sometimes result in unwanted message spam on new week, so kind of mixed feelings. Perhaps alternative approach would be to hide these messages.